schematic/src/x_tabs.c

```diff
@@ -789,14 +790,8 @@ x_tabs_hdr_create (TabInfo* nfo)
 
   /* label:
   */
-  const gchar* fname = NULL;
-  gchar*       bname = NULL;
-
-  if (nfo->page_)
-    fname = s_page_get_filename (nfo->page_);
-
-  if (fname)
-    bname = g_path_get_basename (fname);
+  const gchar* fname = s_page_get_filename (nfo->page_);
+  gchar*       bname = g_path_get_basename (fname);
```

Comment:
I would add a `g_return_val_if_fail` check for **fname** here.

Reply:
Agree.